src/main/java/at/hannibal2/skyhanni/features/inventory/ItemPickupLog.kt

```diff
@@ -0,0 +1,314 @@
+                        }
+
+                        DisplayLayout.ITEM_NAME -> {
+                            add(Renderable.string(name))
+                        }
+
+                        null -> {}
+                    }
+                }
+            },
+        )
+    }
+
+
+    private fun worldChangeCooldown(): Boolean {
+        return LorenzUtils.lastWorldSwitch.passedSince() > 2.seconds
+    }
+
+    private fun updateDisplay() {
+        if (!isEnabled()) return
+
+        val display = mutableListOf<Renderable>()
+
+        itemsAddedToInventory.values.removeIf { it.isExpired() }
+        itemsRemovedFromInventory.values.removeIf { it.isExpired() }
+
+        val removedItemsToNoLongerShow = itemsRemovedFromInventory.toMutableMap()
+        val addedItemsToNoLongerShow = itemsAddedToInventory.toMutableMap()
+
+        if (config.compactLines) {
+            val iterator = addedItemsToNoLongerShow.iterator()
+            while (iterator.hasNext()) {
+                val item = iterator.next()
+
+                if (removedItemsToNoLongerShow.containsKey(item.key)) {
+                    val it = removedItemsToNoLongerShow[item.key]
+                    if (it != null) {
+                        val currentTotalValue = item.value.amount - it.amount
+
+                        if (currentTotalValue > 0) {
+                            display.add(renderList("§a+", currentTotalValue, item.value.name, item.value.neuInternalName))
+                        } else if (currentTotalValue < 0) {
+                            display.add(renderList("§c", currentTotalValue, item.value.name, item.value.neuInternalName))
+                        } else {
+                            itemsAddedToInventory.remove(item.key)
+                            itemsRemovedFromInventory.remove(item.key)
+                        }
+                        removedItemsToNoLongerShow.remove(item.key)
+                        iterator.remove()
+                    }
+                } else {
+                    display.add(renderList("§a+", item.value.amount, item.value.name, item.value.neuInternalName))
+                }
+            }
+        } else {
+            for (item in addedItemsToNoLongerShow) {
+
+                display.add(renderList("§a+", item.value.amount, item.value.name, item.value.neuInternalName))
+                removedItemsToNoLongerShow[item.key]?.let {
+                    display.add(renderList("§c-", it.amount, it.name, it.neuInternalName))
+                    removedItemsToNoLongerShow.remove(item.key)
+                }
+            }
+        }
+        for (item in removedItemsToNoLongerShow) {
+            display.add(renderList("§c-", item.value.amount, item.value.name, item.value.neuInternalName))
+        }
+        val renderable = Renderable.verticalContainer(display, verticalAlign = config.alignment)
+
+        this.display = Renderable.fixedSizeBox(renderable, 30, 75)
```

Comment:
Use a fixed SizeColumn here. So that at least the width is dynamic for the gui editor.
```suggestion
        this.display = Renderable.fixedSizeColumn(renderable, 30)
```

Reply:
![image](https://github.com/user-attachments/assets/91a0a66a-57ff-4ab3-ad4f-c71e9b763098)
The problem is when the item list is empty it will then show up like this. The default time out is 10 seconds which means even if the player picked up an item right before opening the edit gui screen after 10 seconds it would change back to being a narrow column. I agree my current way is flawed, and are open to other ideas but i think it is better than using a column.

Reply:
Don't render an empty list then. Should fix it

Reply:
if i don't render an empty list, if the player hasn't picked up any items the list won't show up at all in the gui editor, is this ok?

Reply:
the gui editor shows gui elements that got rendered recently. this is the same behaviour of every other gui element in skyhanni. we dont want to render/show the gui when its not neeced.